### DISCUSSION TITLE
When to specify `bucket_size` for `jax.ops.segment_sum`?

### DISCUSSION THREAD

**Discussion by n-gao** [jax-ml/jax#12743] [Q&A]
From the documentation of `jax.ops.segment_sum` one can specify a `bucket_size` to improve numerical stability. However, there's little guidance on when to do this and what reasonable values are. So when and how should I pick this value?
Thanks!

**Comment by jakevdp** [comment]
The argument was added here: https://github.com/google/jax/pull/4299 It looks like the intended use-case was for numerical stability when summing a large array of small numbers. @zhangqiaorjc may be able to say more.
